packages/cli/src/ui/components/SettingsDialog.tsx

```diff
@@ -0,0 +1,455 @@
+        setFocusSection('scope');
+      }
+    } else if (focusSection === 'scope') {
+      if (key.upArrow || input === 'k') {
+        if (activeScopeIndex > 0) setActiveScopeIndex(activeScopeIndex - 1);
+      } else if (key.downArrow || input === 'j') {
+        if (activeScopeIndex < scopeItems.length - 1)
+          setActiveScopeIndex(activeScopeIndex + 1);
+      } else if (key.return || input === ' ') {
+        setSelectedScope(scopeItems[activeScopeIndex].value);
+        setFocusSection('settings');
+        setActiveSettingIndex(0);
+        setScrollOffset(0);
+      } else if (key.tab) {
+        setFocusSection('settings');
+      }
+    }
+    if (showRestartPrompt && input === 'r') {
+      // Commit all pending changes to real settings
+      const scope = selectedScope;
+      const pending = pendingSettings;
+      // For each key in pending, set value in settings
+      Object.entries(pending).forEach(([key, value]) => {
+        settings.setValue(scope, key as keyof Settings, value);
+      });
+      setShowRestartPrompt(false);
+      if (onRestartRequest) onRestartRequest();
+    }
+    if (key.escape) {
+      onSelect(undefined, selectedScope);
+    }
+  });
+
+  return (
+    <Box
+      borderStyle="round"
+      borderColor={Colors.Gray}
+      flexDirection="row"
+      padding={1}
+      width="100%"
+      height="100%"
+    >
+      {/* Settings List */}
+      <Box flexDirection="column" flexGrow={1}>
+        <Text bold color={Colors.AccentBlue}>
+          Settings
+        </Text>
+        <Box height={1} />
+        {showScrollUp && <Text color={Colors.Gray}>▲</Text>}
+        {visibleItems.map((item, idx) => {
+          const isActive =
+            focusSection === 'settings' &&
+            activeSettingIndex === idx + scrollOffset;
+          return (
+            <React.Fragment key={item.value}>
+              <Box flexDirection="row" alignItems="center">
+                <Box minWidth={2} flexShrink={0}>
+                  <Text color={isActive ? Colors.AccentGreen : Colors.Gray}>
+                    {isActive ? '●' : ''}
+                  </Text>
+                </Box>
+                <Box minWidth={50}>
+                  <Text
+                    color={isActive ? Colors.AccentGreen : Colors.Foreground}
+                  >
+                    {item.label}
+                  </Text>
+                </Box>
+                <Box minWidth={3} />
+                <Text color={isActive ? Colors.AccentGreen : Colors.Gray}>
+                  {String(item.checked)}
+                </Text>
+              </Box>
+              <Box height={1} />
+            </React.Fragment>
+          );
+        })}
+        {showScrollDown && <Text color={Colors.Gray}>▼</Text>}
+
+        <Box height={1} />
+
+        {/* Apply To Section - Moved to bottom */}
+        <Text color={Colors.AccentBlue}>Apply To</Text>
+        <Box height={1} />
+
+        {/* Display scope items vertically */}
```

Comment:
please align this one with how this UI is rendered in settings. The behavior is currently different  with theme surfacing the selection change immediately while this is a bit different. my preference is to align the two probably to what is in theme and we can later refine this. As Miguel mentioned, our UI for this isn't ideal right now but lets leave polishing that for a leter pr.